app/services/katello/component_view_presenter.rb

```diff
@@ -2,14 +2,26 @@ module Katello
   class ComponentViewPresenter < SimpleDelegator
     attr_accessor :view, :component_view
 
-    def initialize(cv, content_view)
+    def initialize(cv, content_view = nil, component_view = nil)
       @view = content_view
-      @component_view = Katello::ContentViewComponent.where(:composite_content_view_id => cv.id, :content_view_id => @view.id).first_or_initialize
+      component_cv = component_view || Katello::ContentViewComponent.where(:composite_content_view_id => cv.id, :content_view_id => @view.id).first
+      @component_view = component_cv || Katello::ContentViewComponent.where(:composite_content_view_id => cv.id, :content_view_id => @view.id, :latest => true).new
       super(@component_view)
     end
 
-    def self.component_presenter(cv, views:)
-      views.map { |content_view| ComponentViewPresenter.new(cv, content_view) }
+    def self.component_presenter(cv, status, views:)
+      return case status
+             when 'All'
+               views.map { |content_view| ComponentViewPresenter.new(cv, content_view) }
```

Comment:
Could these variables be named to make it more self-explanatory the difference between `cv` and `content_view`? I'm confused..

Reply:
Renamed the variables..composite_cv, component_content_view and content_view_component respectively.

The 3rd is the join object between composite and component and the composite/component terminology is driven from the UI but it shouldn't be too confusing now. Although I am sure I'll be confused looking at this in a month 😄